src/main/scala/piconot/RightHand.scala

```diff
@@ -0,0 +1,36 @@
+package piconot
+
+import java.io.File
+import scalafx.application.JFXApp
+
+import piconot.language._
+
+
+/**
+ * @author dhouck apinson
+ */
+object RightHand extends JFXApp {
+  val rightHandMaze = Maze("resources" + File.separator + "maze.txt")
+
+  // This is the version of the right-hand rule with the fewest states
+  val rulesMinimal = (
+    start state "Start"
+      rule Rclosed turn Left
+      rule Anywhere turn Right go Forward
+  )
+  // This is a more verbose version of the right-hand rule but goes faster
+  val rulesFast = (
+    start state "Start"
+      rule Ropen turn Right go Forward
+      rule (Rclosed, Fopen) go Forward
+      rule (Rclosed, Fclosed, Lopen) turn Left go Forward
+      rule Anywhere turn Back go Forward
+  )
+
```

Comment:
Both implementations are pretty neat. I think you meant fewest rules? I only see one state for both.
It's cool that you can choose to optimize performance or number of rules.